components/brave_federated/task/model.cc

```diff
@@ -0,0 +1,230 @@
+/* Copyright (c) 2023 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#include "brave/components/brave_federated/task/model.h"
+
+#include <algorithm>
+#include <chrono>
+#include <cmath>
+#include <random>
+
+#include "brave/components/brave_federated/task/model_util.h"
+
+namespace brave_federated {
+
+PerformanceReport::PerformanceReport(size_t dataset_size,
+                                     float loss,
+                                     float accuracy,
+                                     std::vector<Weights> parameters,
+                                     std::map<std::string, double> metrics)
+    : dataset_size(dataset_size),
+      loss(loss),
+      accuracy(accuracy),
+      parameters(parameters),
+      metrics(metrics) {}
+
+PerformanceReport::PerformanceReport(const PerformanceReport& other) = default;
+PerformanceReport::~PerformanceReport() = default;
+
+Model::Model(ModelSpec model_spec)
+    : num_iterations_(model_spec.num_iterations),
+      batch_size_(model_spec.batch_size),
+      learning_rate_(model_spec.learning_rate),
+      threshold_(model_spec.threshold) {
+  std::random_device rd;
+  std::mt19937 mt(rd());
+  std::uniform_int_distribution<> distribution(-10.0, 10.0);
```

Comment:
@atuchin-m given that `<random>` is banned, what would be the best way to sample from a uniform distribution? I see we have some internal random lib in `brave_base` but it might be an overshot for this use case

Reply:
also `absl::uniform_int_distribution` seems to be banned